CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.16)
```

Comment:
Please set to the lowest still supported version which would be 3.5.

Reply:
I don't think this is necessary as this project does not rely on any fancy CMake feature.

Reply:
It would be better to use the range notation.  The minimum version doesn't even need to change if you don't want. (i.e. `3.1...3.27` should work.  As an aside: I'm not certain but I believe there shouldn't be spaces between the two versions since the syntax was added in a way to be backwards compatible with ancient versions of CMake from before the range feature as added.)  The advantage to using the range notation is that you're not turning on unnecessary compatibility features and that cleans up and improves the ability to vendor the library with add_subdirectory.

Reply:
I suppose I should follow the owner's (@nlohmann) request of using the minimum supported of 3.5 instead of a range. It seems the test scripts use cmake 3.13, so that too should resolve it. Let me know if 3.13 is preferred otherwise I will edit the pull request with 3.5 as requested.

Reply:
What happens when the range is given as `3.1...3.27` and I am running CMake 3.50?

Reply:
If you are running CMake 3.50 the script will work as long as CMake 3.27 is still supported (like CMake 3.5 is supported by CMake 3.27). 

Basically the range notation means: "I require at least CMake 3.1 and I verified that still works with CMake 3.27. So if you are using CMake 3.50, please use the CMake 3.27 compatibility mode."

Reply:
Not using the range notation is the same as using `3.1...3.1` as the range notation.

Reply:
Yeah, to provide a little more technical detail: the range notation just means that cmake_policy introduced between CMake versions min and max are to be set to NEW instead of OLD.  The reason for the new warning about < 3.5 support being dropped is not that Kitware cares that you're writing scripts compatible with < 3.5 but rather that they're removing the OLD policies and the script might break if it hasn't been validated to work with the policies set to NEW.  I can't think of any reason for a project since the feature was introduced to not use the range notation.

Reply:
I have revised the commit on this PR to the range notation. I have tested it with cmake version 3.5 and cmake version 3.27 installed on my system. No additional warnings seen for either one. 

Here's a capture of the output:


>###  3.27
> /workspaces/json (update-min-cmake-version) $ cmake --version
> cmake version 3.27.0
> 
> CMake suite maintained and supported by Kitware (kitware.com/cmake).
>  /workspaces/json (update-min-cmake-version) $ cd build/
>  /workspaces/json/build (update-min-cmake-version) $ cmake ..
> -- The CXX compiler identification is GNU 9.4.0
> -- Detecting CXX compiler ABI info
> -- Detecting CXX compiler ABI info - done
> -- Check for working CXX compiler: /usr/bin/c++ - skipped
> -- Detecting CXX compile features
> -- Detecting CXX compile features - done
> -- Using the multi-header code from /workspaces/json/include/
> -- Found Git: /usr/local/bin/git (found version "2.41.0") 
> -- Operating system: Linux-5.15.0-1041-azure; Distributor ID:   Ubuntu; Description:    Ubuntu 20.04.6 LTS; Release:    20.04; Codename:        focal; Linux codespaces-2fc995 5.15.0-1041-azure #48-Ubuntu SMP Tue Jun 20 20:34:08 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
> -- Compiler: c++ (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0; Copyright (C) 2019 Free Software Foundation, Inc.; This is free software; see the source for copying conditions.  There is NO; warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
> -- Testing standards: 11 14 17 20 [23]
> -- Looking for C++ include sys/types.h
> -- Looking for C++ include sys/types.h - found
> -- Looking for C++ include stdint.h
> -- Looking for C++ include stdint.h - found
> -- Looking for C++ include stddef.h
> -- Looking for C++ include stddef.h - found
> -- Check size of size_t
> -- Check size of size_t - done
> -- Configuring done (2.4s)
> -- Generating done (0.2s)
> -- Build files have been written to: /workspaces/json/build
> 
> 
>###  3.5 
> /workspaces/json (update-min-cmake-version) $ sudo cmake --version
> cmake version 3.5.0
> 
> CMake suite maintained and supported by Kitware (kitware.com/cmake).
> /workspaces/json (update-min-cmake-version) $ cd build/
>  /workspaces/json/build (update-min-cmake-version) $ sudo cmake ..
> -- The CXX compiler identification is GNU 9.4.0
> -- Check for working CXX compiler: /usr/bin/c++
> -- Check for working CXX compiler: /usr/bin/c++ -- works
> -- Detecting CXX compiler ABI info
> -- Detecting CXX compiler ABI info - done
> -- Detecting CXX compile features
> -- Detecting CXX compile features - done
> -- Using the multi-header code from /workspaces/json/include/
> -- Configuring done
> -- Generating done
> -- Build files have been written to: /workspaces/json/build
>